kaldb/src/main/java/com/slack/kaldb/chunkManager/ChunkManager.java

```diff
@@ -68,7 +75,10 @@ private static ExecutorService queryThreadPool() {
             .map(
                 (chunk) ->
                     CompletableFuture.supplyAsync(
-                            () -> chunk.query(query),
+                            () -> {
+                              Tracing.current().currentTraceContext().maybeScope(context);
+                              return chunk.query(query);
+                            },
```

Comment:
We should also clear the context after the query here. Prefer to do it a finally block so it's more robust. Otherwise, if some other function uses the same thread pool we will get weird traces and those issues are hard to debug..